web/misc.go

```diff
@@ -168,12 +168,19 @@ func parseIntoBSO(jsonData json.RawMessage, bso *syncstorage.PutBSOInput) *parse
 
 	if r, ok := bkeys["ttl"]; ok {
 		var ttl int
-		err := json.Unmarshal(r, &ttl)
-		if err != nil {
-			return &parseError{bId: bId, field: "ttl", msg: "Invalid format"}
+
+		// bug 1332552, treat `ttl:null` as never expiring
+		// by using 100 years for it (in seconds)
+		if len(r) == 4 && string(r[:]) == "null" {
```

Comment:
Sometimes just dealing with it as a []byte is easier. JSON schemas where `{"x":123}`, `{"x":null}` or `{"x":"123"}` is all valid data from clients makes me sad. It's really difficult to work with it in golang cause you want to unmarshal the into structs. 

When json.Unmarshal tries to parse `nu`, `"null"` or `[]` into an `int` type [it'll fail](https://play.golang.org/p/3QeYyUPDlO). That will bubble up into an HTTP 400 for the client.